### PR TITLE
feat: instrument go-redis clients (v7, v8)

### PR DESCRIPTION
**Pull request by RomainMuller**
Add instrumentation logic for `go-redis/redis` clients (v7 and v8).

### CODE REVIEW

File: internal/injector/injector.go
[the remark this answers is not in the document]
Comment (RomainMuller) [reply internal/injector/injector.go:1]: Found a bug & fixed it locally... I guess I could extract that underneath this PR.

File: .github/workflows/generate.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ jobs:
           cache-dependency-path: '**/*.sum'
       - name: Run 'go generate ./...'
         run: go generate ./...
+      - name: Run 'go mod tidy'
+        run: find . -iname go.mod | xargs dirname | xargs -I DIR go -C DIR mod tidy
```
Comment (eliottness) [inline .github/workflows/generate.yml:28]: Does this work: `find -name go.mod -execdir go mod tidy \;`?
it's the same we have on dd-trace-go

File: _integration-tests/tests/go-redis.v7/validation.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          "resource": "get",
+          "type": "redis",
+          "meta": {
+            "redis.args_length": "2",
+            "component": "go-redis/redis.v7",
+            "out.db": "0",
+            "span.kind": "client",
+            "db.system": "redis",
+            "redis.raw_command": "get test_key: ",
+            "out.host": "localhost"
```
Comment (eliottness) [inline _integration-tests/tests/go-redis.v7/validation.json:27]: Will this work even with the NAT driver used on windows?

**Review by eliottness — APPROVED** [review]
Approved except for the Powershell :stuck_out_tongue:

File: _integration-tests/tests/go-redis.v7/validation.json (hunk 2 above)
Comment (eliottness) [reply _integration-tests/tests/go-redis.v7/validation.json:27]: I can try this at home if you need

File: .github/workflows/generate.yml (hunk 1 above)
Comment (RomainMuller) [reply .github/workflows/generate.yml:28]: Duh I was not aware of `-execdir` (I haven't used this in a while, to be fair)

**Pull request merged**